mongo/mongo.go

```diff
@@ -435,26 +441,38 @@ func EnsureServer(args EnsureServerParams) (Version, error) {
 
 func ensureServer(args EnsureServerParams, mongoKernelTweaks map[string]string) (Version, error) {
 	var zeroVersion Version
+	var err error
 	tweakSysctlForMongo(mongoKernelTweaks)
 	logger.Infof(
 		"Ensuring mongo server is running; data directory %s; port %d",
 		args.DataDir, args.StatePort,
```

Comment:
So with the check below, data dir here is wrong, so this log Info really needs to be part of the "if using snap" conditional below